### PR TITLE
qt: 5.12.0

### PR DESCRIPTION
**Pull request by scpeters**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by Fohlen** [comment]
Since `5.12` has reached `final release` candidate and will bring `WebAssembly` support to QT it would be appreciated to have this air in homebrew as soon as possible 🥂

**Review by Fohlen — APPROVED** [review]
Seems fine to me 👍

File: Formula/qt.rb [hunk 1]
```diff
@@ -3,33 +3,13 @@
...
-    # https://bugs.chromium.org/p/chromium/issues/detail?id=840251
-    # https://bugs.chromium.org/p/chromium/issues/detail?id=849689
-    patch do
-      url "https://raw.githubusercontent.com/Homebrew/formula-patches/962f0f/qt/xcode10.diff"
-      sha256 "c064398411c69f2e1c516c0cd49fcd0755bc29bb19e65c5694c6d726c43389a6"
-    end
+    url "https://download.qt.io/official_releases/qt/5.12/5.12.0/single/qt-everywhere-src-5.12.0.tar.xz"
+    mirror "https://qt.mirror.constant.com/archive/qt/5.12/5.12.0/single/qt-everywhere-src-5.12.0.tar.xz"
+    mirror "https://ftp.osuosl.org/pub/blfs/conglomeration/qt5/qt-everywhere-src-5.12.0.tar.xz"
+    sha256 "356f42d9087718f22f03d13d0c2cdfb308f91dc3cf0c6318bed33f2094cd9d6c"
```
Comment (igas) [inline Formula/qt.rb]: As there are no patches anymore, could you please delete `stable` block and put this on top under `homepage`

Comment (Fohlen) [reply Formula/qt.rb]: @igas I do not know if this is useful. According to the [release schedule](https://wiki.qt.io/Qt_5.12_Release) QT will release patches in the near future (depending on what's wrong). So tagging this as stable is technically the most correct :)

**Comment by igas** [comment]
@fxcoudert I think we should kill options as per https://github.com/Homebrew/homebrew-core/issues/31510 there almost no installs of them.

For 365 days:
```
#87   qt                                                            405,476 0.27%
#4238 qt --with-examples                                                343 0.00%
#4410 qt --with-docs --with-examples --with-mysql --with-postgresql     309 0.00%
#4614 qt --with-docs --with-examples                                    280 0.00%
#4672 qt --HEAD                                                         271 0.00%
#4806 qt --with-mysql                                                   252 0.00%
#5009 qt --with-docs                                                    230 0.00%
#5588 qt --with-mysql-client                                            173 0.00%
#5854 qt --with-examples --with-mysql-client --with-postgresql          156 0.00%
#5870 qt --with-postgresql                                              155 0.00%
#6180 qt --with-mysql --with-postgresql                                 134 0.00%
#6181 qt --with-mysql-client --with-postgresql                          134 0.00%
#7001 qt --with-docs --with-mysql --with-postgresql                      97 0.00%
#7192 qt --with-examples --with-postgresql                               91 0.00%
#7415 qt --with-docs --with-examples --with-mysql                        84 0.00%
#7943 qt --with-docs --with-examples --with-postgresql                   71 0.00%
#9143 qt --with-docs --with-mysql                                        50 0.00%
#9829 qt --with-examples --with-mysql-client                             42 0.00%
```

**Comment by SMillerDev** [comment]
install_on_request might be a better metric but I think you're right. @scpeters could you remove the options?

File: Formula/qt.rb (hunk 1 above)
Comment (scpeters) [reply Formula/qt.rb]: the audit didn't complain, so I'm inclined to leave it as is

**Comment by scpeters** [comment]
> install_on_request might be a better metric but I think you're right. @scpeters could you remove the options?

options removed in 10d4371da612fcb1d5f387a2153431aec9cfb90d

**Review by MarcelRaschke — APPROVED** [review]

File: Formula/qt.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/qt.rb]: Normally we'd remove it, unless we know patches we will re-added soon (but it's not the case here). But given how long Qt build is, we probably only want to do that when we push another commit to it (see below).

File: Formula/qt.rb [hunk 2]
```diff
@@ -61,31 +34,13 @@ def install
...
-        else
-          exec mysql_config "$@"
-        fi
-      EOS
-      chmod 0755, "brew_shim/mysql_config"
-      args << "-mysql_config" << buildpath/"brew_shim/mysql_config"
-    end
-
-    args << "-plugin-sql-psql" if build.with? "postgresql"
-    args << "-proprietary-codecs" if build.with? "proprietary-codecs"
```
Comment (fxcoudert) [inline Formula/qt.rb]: I think you should add back `-proprietary-codecs` which was default behaviour before (the option was `--without-`).

File: Formula/qt.rb (hunk 1 above)
Comment (scpeters) [reply Formula/qt.rb]: sounds good, it's now removed in e6a57c3

File: Formula/qt.rb (hunk 2 above)
Comment (scpeters) [reply Formula/qt.rb]: thanks for catching that! I've added it back and squashed in 8c2e33f

**Comment by fxcoudert** [comment]
Looking good… some minor details, which can be fixed when we merge:
- `qt: 5.12.0` should be `qt 5.12.0` for version update (no colon)
- when we update version and change other stuff, we do a single commit (per formula); in that case, the commit message is `qt 5.12.0`, because version update trumps other changes :)

**Review by fxcoudert — APPROVED** [review]

**Comment by igas** [comment]
I think liblastfm and qmmp need revision bumps

**Comment by fxcoudert** [comment]
@igas I don't think they do, the broken linkage is not to Qt libs, but to their own libraries. It's a semi-false positive, where the install name of the library is wrong. It's preexisting to this PR. It doesn't prevent the software from working (as the tests pass), but could lead to trouble in some cases.

**Pull request closed without merging**